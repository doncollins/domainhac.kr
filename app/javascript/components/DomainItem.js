import React from 'react'
import { formatRelative } from 'date-fns'

export default function({ domain }) {
    const formattedDate = formatRelative(
        new Date(domain.availability.updated_at),
        new Date()
    )

    return (
        <div className="px-2 py-1 w-full flex items-center justify-between bg-gray-100 even:bg-white">
            <i className="fa fa-check text-blue-500" />

            <div className="flex-grow ml-8 sm:ml-2 flex items-center justify-start">
                <div className="sm:w-32 text-right">{domain.label}</div>
                <div className="text-blue-500">.</div>
                <div>{domain.tld.label}</div>
            </div>

            <div className="text-xs text-gray-500 flex items-center justify-end sm:w-1/2">
                <span class="hidden sm:inline">Updated&nbsp;</span>
                <span>{formattedDate}</span>
            </div>
        </div>
    )
}