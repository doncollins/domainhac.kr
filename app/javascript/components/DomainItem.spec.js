import React from 'react'
import renderer from 'react-test-renderer';
import mockNow from '../test/mockNow';
import DomainItem from './DomainItem';

it('renders correctly', () => {
    mockNow()

    const domain = {
        id: 1,
        label: 'f',
        tld: {
            id: 1,
            label: 'oo',
        },
        availability: {
            id: 1,
            available: true,
            updated_at: '2021-06-05T22:14:11+00:00',
        },
    }

    const component = renderer.create(
        <DomainItem domain={domain} />
    )

    expect(component.toJSON()).toMatchSnapshot()
})