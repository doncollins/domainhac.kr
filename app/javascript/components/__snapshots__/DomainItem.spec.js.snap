// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="px-2 py-1 w-full flex items-center justify-between bg-gray-100 even:bg-white"
>
  <i
    className="fa fa-check text-blue-500"
  />
  <div
    className="flex-grow ml-8 sm:ml-2 flex items-center justify-start"
  >
    <div
      className="sm:w-32 text-right"
    >
      f
    </div>
    <div
      className="text-blue-500"
    >
      .
    </div>
    <div>
      oo
    </div>
  </div>
  <div
    className="text-xs text-gray-500 flex items-center justify-end sm:w-1/2"
  >
    <span
      class="hidden sm:inline"
    >
      UpdatedÂ 
    </span>
    <span>
      yesterday at 10:14 PM
    </span>
  </div>
</div>
`;
